### PR TITLE
Fix build on OCaml 4.06.1

### PR DESCRIPTION
**Pull request by Julow**
Unrelated to https://github.com/ocaml-ppx/ocamlformat/pull/642 but since it was merged, build was broken on my machine.

### CODE REVIEW

File: src/Source.ml [hunk 1]
```diff
@@ -92,9 +92,9 @@ let extend_loc_to_include_attributes t (loc : Location.t)
           let loc =
             match payload with
             | PStr [] -> loc
-            | PStr l -> (List.last_exn l).pstr_loc
+            | PStr l -> (List.last_exn l).Parsetree.pstr_loc
```
Comment (Julow) [inline src/Source.ml]: This was just a warning.

File: test/passing/js_source.ml.ref [hunk 2]
```diff
@@ -9792,4 +9792,4 @@ module Indexop = struct
   let x, y, z = Def.(h.%["one"], h.%("two"), h.%{"three"})
 end
 
-type t = |
+(* type t = | *)
```
Comment (Julow) [inline test/passing/js_source.ml.ref]: This change may not be required, we can just assume that we are using 4.07 when running tests ?

Comment (gpetiot) [reply test/passing/js_source.ml.ref]: So you were using 4.06.1, didn't you already get errors on this construct?

Comment (Julow) [reply test/passing/js_source.ml.ref]: No.

Comment (gpetiot) [reply test/passing/js_source.ml.ref]: Do you have an idea of what's happening? Why do we have to comment this now? It causes an error on 4.06.1 I don't understand why you didn't have this error previously.

**Comment by hhugo** [comment]
ocamlformat compiles/installs fine with ocaml 4.06.0

```
$ ocamlc -version
4.06.1
$ make clean
rm -Rf _coverage
find ./ -name 'bisect*.out' -delete
rm -rf _build dune-workspace
$ dune build -p ocamlformat
$
```
note that call `make` will fail because more warning are enabled in that case.

**Comment by hhugo** [comment]
`ocamlformat_reason` doesn't work and this PR fixes this.

File: src/Source.ml (hunk 1 above)
Comment (hhugo) [reply src/Source.ml]: I think it's fine to have warning trigger with older version of ocaml, @jberdine, any opinion ?

File: test/passing/js_source.ml.ref (hunk 2 above)
Comment (hhugo) [reply test/passing/js_source.ml.ref]: I would not expect tests to pass with older version of the compiler.

Comment (jberdine) [reply test/passing/js_source.ml.ref]: Yes, there are too many moving parts to expect tests to pass with old compilers. My position is just one, heavily influenced by a development context that is perhaps uncommon, and perhaps not widely shared, but I don't think it's worth the time to worry about 4.06 when 4.08 is about to be released. I am much less worried about requiring a current compiler to build ocamlformat than I am about generating code that only the current compiler can parse, but that is a different question. For building ocamlformat itself, I'd say just only develop using the current compiler (to save your own sanity by avoiding chasing down the consequences of the differences between the parsers in different versions) and bump the minimum required version.

Comment (Julow) [reply test/passing/js_source.ml.ref]: I agree for the tests, I removed the commit.

**Comment by Julow** [comment]
I think this is ready to be merged.

**Pull request merged**